internal/webhooks/cluster.go

```diff
@@ -110,10 +110,25 @@ func (webhook *Cluster) Default(ctx context.Context, obj runtime.Object) error {
 			return apierrors.NewInternalError(errors.Wrapf(err, "Cluster %s can't be defaulted. ClusterClass %s can not be retrieved", cluster.Name, cluster.Spec.Topology.Class))
 		}
 
+		// Validate cluster class variables transitions that may be enforced by CEL validation rules on variables.
+		// If no request found in context, then this has not come via a webhook request, so skip validation of old cluster.
+		var oldCluster *clusterv1.Cluster
+		req, err := admission.RequestFromContext(ctx)
```

Comment:
Are there non-nil error cases that need to be accounted for?

Reply:
Added returning non-wrapped error following pattern in other defaulting webhooks in this project.

Reply:
Actually, as described in the comment, this defaulting func is also called from other places where `req` may not be set in the context. That is valid and the only error that can actually be returned here. In that case we skip reading the old cluster.

Reply:
Let's also provide the request in the context in these unit tests, then we don't have to handle that case that cannot happen in "regular" runtime

Reply:
I'll take another look

Reply:
We could adjust the unit tests to fit to reality. At runtime this is never called without the request by CR. But the safeguard is fine, resolving here